app/src/main/java/org/oppia/app/parser/StringToFractionParser.kt

```diff
@@ -53,4 +99,22 @@ class StringToFractionParser {
   }
 
   private fun isInputNegative(inputText: String): Boolean = inputText.startsWith("-")
+  /** Enum to store the errors of [FractionInputInteractionView]. */
+  enum class FractionParsingError(@StringRes error: Int?) {
+    VALID(error = null),
+    INVALID_CHARS(error = R.string.fraction_error_invalid_chars),
+    INVALID_FORMAT(error = R.string.fraction_error_invalid_format),
+    DIVISION_BY_ZERO(error = R.string.fraction_error_divide_by_zero);
+
+    private var error: Int?
+
+    init {
+      this.error = error
+    }
```

Comment:
Above 4 lines can be removed and the constructor can be updated to
`enum class FractionParsingError(@StringRes private var error: Int?) {`

Reply:
Updated